jax_dataclasses/_copy_and_mutate.py

```diff
@@ -15,20 +15,25 @@ class _Mutability(enum.Enum):
     MUTABLE_NO_VALIDATION = enum.auto()
 
 
-def _mark_mutable(obj: Any, mutable: _Mutability) -> None:
+def _mark_mutable(obj: Any, mutable: _Mutability, visited = set()) -> None:
```

Comment:
It seems like using the (mutable) `set()` as the default here is going to cause issues, since every call to `_mark_mutable()` is going to share the same visited set.

I'd expect with the current change that objects are correctly unfrozen at the start of the context manager, but will never be refrozen (since they'll be in the shared visited set); is this true?

If so — maybe this can instead be written as `visited: Optional[Set[Any]]=None`? And initialized in the function if `visited is None`; alternatively, a default can just not be specified and instead `visited` can be instantiated / passed in wherever `_mark_mutable` is first called.